### PR TITLE
CMP-2524: Only load node profiles for managed OpenShift

### PR DESCRIPTION
**Pull request by rhmdnd**
Implement support for only loading node profiles on managed offerings
since the control plane isn't accessible. This reduces the amount of
confusion for users by only presenting profiles that can be run on the
platform.

### CODE REVIEW

**Comment by rhmdnd** [comment]
This still has a rough edge with the kustomization work and detecting platform deployments. Need to find an orthogonal way to handle spinning up tests on platforms.

**Comment by rhmdnd** [comment]
The ROSA test worked... but still need to do some cleanup

```
2024/05/15 03:34:08 ProfileBundle ready (VALID)
2024/05/15 03:34:13 ProfileBundle ready (VALID)
bypassing MachineConfigPool test setup because it's not supported on rosa
bypassing MachineConfigPool test setup because it's not supported on rosa
=== RUN   TestInstallOnlyParsesNodeProfiles
=== PAUSE TestInstallOnlyParsesNodeProfiles
=== CONT  TestInstallOnlyParsesNodeProfiles
--- PASS: TestInstallOnlyParsesNodeProfiles (0.04s)
PASS
2024/05/15 03:34:18 cleaning up Profile Bundles
```

File: bundle/manifests/compliance-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -1661,4 +1661,5 @@ spec:
     name: operator
   - image: ghcr.io/complianceascode/k8scontent:latest
     name: profile
+  replaces: 1.3.1
```
Comment (rhmdnd) [inline bundle/manifests/compliance-operator.clusterserviceversion.yaml:1664]: Bah - this can get ripped out. It's an unnecessary change.

**Review by Vincent056 — APPROVED** [review]
